kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/ConvertUtils.java

```diff
@@ -81,24 +97,45 @@ public static ProcessInstanceList convertToProcessInstanceWithVarsList(Collectio
             return new ProcessInstanceList(new org.kie.server.api.model.instance.ProcessInstance[0]);
         }
 
-        List<ProcessInstance> processInstances = new ArrayList<ProcessInstance>(instances.size());
+        List<ProcessInstance> processInstances = new ArrayList<>(instances.size());
         for (ProcessInstanceWithVarsDesc pi : instances) {
             org.kie.server.api.model.instance.ProcessInstance instance = convertToProcessInstance(pi);
-
-            instance.setVariables(pi.getVariables());
-
+            if (instance != null) {
+                instance.setVariables(pi.getVariables());
+            }
             processInstances.add(instance);
         }
-
         return new ProcessInstanceList(processInstances);
     }
 
     public static org.kie.server.api.model.instance.ProcessInstance convertToProcessInstance(ProcessInstanceDesc pi) {
+        return convertToProcessInstance(pi, null);
+    }
+
+    private static void checkContainerId(long processInstanceId, String containerId) {
+        // check process instance belong to container id
+        if (containerId != null) {
+            try {
```

Comment:
Im wondering if rather than checking that process id instance is contained by the container here, we should add containerId parameter to RuntimeDataService methods that accepts processInstance, that will avoid a second query and is cleaner. 
Wdty?

Reply:
do you mean modifying, for instance, the one below?

https://github.com/kiegroup/jbpm/blob/master/jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/RuntimeDataServiceImpl.java#L447

To me it will be clearer and cleaner but I guess it will be a wider change at the same time since it may mean to modify the interface.

Reply:
It is definitely an interface change ;), that why Im not implementing it right away. Lets wait for @elguardian opinion